db/migrate/20241203082541_datasets_retirement.rb

```diff
@@ -0,0 +1,439 @@
+    BU19270200 BU19270201 BU19270202 BU19270300 BU19270309 BU19270400 BU19270409 BU19270500 BU19270501 BU19270600
+    BU19270609 BU19270700 BU19270800 BU19270801 BU19270802 BU19270803 BU19270900 BU19270901 BU19270909 BU19271000
+    BU19271100 BU19271109 BU19300101 BU19300102 BU19300103 BU19300104 BU19300105 BU19300106 BU19300201 BU19300202
+    BU19300301 BU19300302 BU19300401 BU19300501 BU19300502 BU19300503 BU19300504 BU19300601 BU19300602 BU19300603
+    BU19300701 BU19300702 BU19300703 BU19300801 BU19300802 BU19300803 BU19300804 BU19300805 BU19300901 BU19300902
+    BU19300903 BU19300904 BU19300905 BU19301001 BU19301002 BU19301003 BU19301004 BU19301101 BU19301102 BU19301201
+    BU19301202 BU19301203 BU19301204 BU19301205 BU19301206 BU19301301 BU19301401 BU19301402 BU19301403 BU19301404
+    BU19301405 BU19301406 BU19301407 BU19301408 BU19301409 BU19301410 BU19301411 BU19301501 BU19301502 BU19301503
+    BU19301504 BU19301505 BU19301506 BU19301601 BU19301701 BU19301702 BU19301703 BU19301704 BU19301705 BU19301706
+    BU19301707 BU19301708 BU19302501 BU19302502 BU19302503 BU19302504 BU19302701 BU19302702 BU19302703 BU19302901
+    BU19302902 BU19303101 BU19303102 BU19303103 BU19303104 BU19303105 BU19303301 BU19303302 BU19303501 BU19303502
+    BU19303503 BU19305001 BU19305002 BU19305003 BU19310000 BU19310001 BU19310002 BU19310003 BU19310004 BU19310005
+    BU19310006 BU19310007 BU19310008 BU19310009 BU19310010 BU19310011 BU19310100 BU19310101 BU19310102 BU19310103
+    BU19310104 BU19310105 BU19310106 BU19310107 BU19310108 BU19310109 BU19310200 BU19310208 BU19310209 BU19310300
+    BU19310400 BU19310401 BU19310402 BU19310403 BU19310404 BU19310409 BU19310500 BU19310501 BU19310502 BU19310503
+    BU19310504 BU19310600 BU19310700 BU19310701 BU19310800 BU19310801 BU19310802 BU19310803 BU19310804 BU19310805
+    BU19310806 BU19310807 BU19310808 BU19310809 BU19310810 BU19311000 BU19311001 BU19311002 BU19311003 BU19311004
+    BU19311005 BU19311006 BU19311007 BU19311008 BU19311100 BU19311101 BU19311200 BU19311201 BU19311202 BU19311203
+    BU19311204 BU19311205 BU19660506
+  ].freeze
+
+  def up
+    # Counting and printing total number of datasets to be destroyed
+    total_keys = GEO_IDS_COUNTRIES.size + GEO_IDS_MUNICIPALITIES.size + GEO_IDS_REGIONS.size + GEO_IDS_NEIGHBOURHOODS.size
+    puts "Number of datasets to be destroyed: #{total_keys}. Start destroying datasets, may take a while."
+
+    @deleted_datasets = []
+
+    destroyed_count = 0
+
+    # Process each category of GEO_IDS
+    destroyed_count += process_geo_ids(GEO_IDS_COUNTRIES)
+    destroyed_count += process_geo_ids(GEO_IDS_MUNICIPALITIES)
+    destroyed_count += process_geo_ids(GEO_IDS_REGIONS)
+    destroyed_count += process_geo_ids(GEO_IDS_NEIGHBOURHOODS)
+
+    # Store the deleted records for rollback purposes
+    save_deleted_datasets
+
+    # Print number of actually destroyed datasets
+    puts "Destroyed #{destroyed_count} datasets in total."
```

Comment:
```suggestion
    all_ids = GEO_IDS_COUNTRIES + GEO_IDS_MUNICIPALITIES + GEO_IDS_REGIONS + GEO_IDS_NEIGHBOURHOODS
    deleted = Dataset.where(geo_id: all_ids).destroy_all
    save_deleted_datasets(deleted.to_json)
    
    puts "Destroyed #{deleted.count} datasets in total."
```
Something like this might be a bit quicker, if you find your code being slow!
I'll check it better on Thursday!